src/test/java/seedu/address/model/tag/TagTest.java

```diff
@@ -19,8 +21,16 @@ public void constructor_invalidTagName_throwsIllegalArgumentException() {
 
     @Test
     public void isValidTagName() {
-        // null tag name
+        //Invalid Tag Names
         assertThrows(NullPointerException.class, () -> Tag.isValidTagName(null));
+        assertFalse(Tag.isValidTagName("")); // empty string
+        assertFalse(Tag.isValidTagName("Tag_That_Has_More_Than_20_Characters")); // more than 30 characters
+        assertFalse(Tag.isValidTagName("Tag_That_Has More_Than_20_Characters")); // more than 30 characters
```

Comment:
Is line 27 repeated with line 28?

Reply:
28 has the empty space, i'll make it clearer in the comments